core/domain/summary_services_test.py

```diff
@@ -879,3 +879,147 @@ def test_requiring_public_activities_to_be_public_succeeds(self):
                 feconf.ACTIVITY_TYPE_EXPLORATION, self.EXP_ID_0),
             activity_domain.ActivityReference(
                 feconf.ACTIVITY_TYPE_COLLECTION, self.COL_ID_2)])
+
+
+class CollectionNodeMetadataDictsTest(
+        exp_services_test.ExplorationServicesUnitTests):
+    """Test functions for getting collection node metadata dicts."""
+
+    ALBERT_EMAIL = 'albert@example.com'
+    ALBERT_NAME = 'albert'
+
+    BOB_EMAIL = 'bob@example.com'
+    BOB_NAME = 'bob'
+
+    EXP_ID1 = 'eid1'
+    EXP_ID2 = 'eid2'
+    EXP_ID3 = 'eid3'
+    EXP_ID4 = 'eid4'
+    EXP_ID5 = 'eid5'
+    INVALID_EXP_ID = 'invalid_exp_id'
+
+    def setUp(self):
+        super(CollectionNodeMetadataDictsTest, self).setUp()
+
+        self.albert_id = self.get_user_id_from_email(self.ALBERT_EMAIL)
+        self.signup(self.ALBERT_EMAIL, self.ALBERT_NAME)
+        self.bob_id = self.get_user_id_from_email(self.BOB_EMAIL)
+        self.signup(self.BOB_EMAIL, self.BOB_NAME)
+
+        self.save_new_valid_exploration(self.EXP_ID1, self.albert_id,
+                                        title='Exploration 1 Albert title',
+                                        objective='An objective 1')
+
+        self.save_new_valid_exploration(self.EXP_ID2, self.albert_id,
+                                        title='Exploration 2 Albert title',
+                                        objective='An objective 2')
+
+        self.save_new_valid_exploration(self.EXP_ID3, self.albert_id,
+                                        title='Exploration 3 Albert title',
+                                        objective='An objective 3')
+
+        self.save_new_valid_exploration(self.EXP_ID4, self.bob_id,
+                                        title='Exploration 4 Bob title',
+                                        objective='An objective 4')
+
+        self.save_new_valid_exploration(self.EXP_ID5, self.albert_id,
+                                        title='Exploration 5 Albert title',
+                                        objective='An objective 5')
+
+        rights_manager.publish_exploration(self.albert_id, self.EXP_ID1)
+        rights_manager.publish_exploration(self.albert_id, self.EXP_ID2)
+        rights_manager.publish_exploration(self.albert_id, self.EXP_ID3)
+        rights_manager.publish_exploration(self.bob_id, self.EXP_ID4)
+
+        exp_services.index_explorations_given_ids([
+            self.EXP_ID1, self.EXP_ID2, self.EXP_ID3,
+            self.EXP_ID4])
+
+    def test_get_exploration_metadata_dicts(self):
+        summaries = (summary_services.get_exploration_metadata_dicts(
```

Comment:
These are not summaries; they're metadata dicts. Please name them accordingly to avoid confusion. Ditto for expected_summaries below, and elsewhere.